Storage/SQL/BookmarksSqlite.swift

```diff
@@ -105,17 +105,19 @@ class BookmarkTable<T> : GenericTable<BookmarkNode> {
 
     override var factory: ((row: SDRow) -> BookmarkNode)? {
         return { row -> BookmarkNode in
-            let bookmark = BookmarkItem(guid: row["guid"] as String,
-                title: row["title"] as String,
-                url: row["bookmarkUrl"] as String)
+            let url = row["bookmarkUrl"] as! String
+            let guid = row["guid"] as! String
+            let bookmark = BookmarkItem(guid: guid,
+                                        title: row["title"] as? String ?? url,
```

Comment:
Are we sure we want to do this substitution at this low of a level? Seems like this might hide bugs. Are there any non-exceptional situations where the cast to String will fail?